Assets/Scripts/Tank/TankService.cs

```diff
@@ -1,37 +1,97 @@
-﻿using UnityEngine;
+﻿using System.Collections.Generic;
+using UnityEngine;
 public class TankService : MonoSingleton<TankService>
 {
-    public TankView tankView;
     private Camera mainCamera;
-    private TankModel tankModel;
-    private TankController tankController;
-    protected Joystick stick;
+
+    [SerializeField]
+    private List<EnemyController> enemyList;
+    [SerializeField]
+    private List<Transform> spawnPositions;
+    [SerializeField]
+    private SpawnerObject spawnerObject;
+    public List<TankScriptableObject> enemiesToGenerate;
+    public Transform enemyParent;
+    public EnemyController redTank;
+    public EnemyController blueTank;
+    public EnemyController greenTank;
+
+    [SerializeField]
+    private TankController playerTank;
+
+    public Transform PlayerTransform{
+        get{
+            Transform playerTransform = playerTank.transform;
+            return playerTransform;
+        }
+    }
+
     private void Start() {
-        GetJoystick();
-        CreateTank();
-        AssignCamera();
+        populateSpawnPositions();
+        CreatePlayer();
+        EnemyController createdEnemy;
+        createdEnemy = CreateEnemy(enemiesToGenerate[0]);
+        enemyList.Add(createdEnemy);
+        createdEnemy = CreateEnemy(enemiesToGenerate[1]);
+        enemyList.Add(createdEnemy);
+        createdEnemy = CreateEnemy(enemiesToGenerate[2]);
+        enemyList.Add(createdEnemy);
+    }
+
+    // Player Creation...
+    private void CreatePlayer(){
+        Vector3 randomSpawnPosition = GetRandomSpawnPos();
+        playerTank.transform.position = randomSpawnPosition;
```

Comment:
Instantiate Player tank using prefabs instead of already having one in the scene

Reply:
done